Modules/AddonUpdateNotifier/AddonUpdateNotifier.lua

```diff
@@ -0,0 +1,60 @@
+---@class PvPAssistant
+local PvPAssistant = select(2, ...)
+local addonName = ...
+
+local GUTIL = PvPAssistant.GUTIL
+
+---@class PvPAssistant.ADDON_UPDATE_NOTIFIER : Frame
+PvPAssistant.ADDON_UPDATE_NOTIFIER = GUTIL:CreateRegistreeForEvents({ "PLAYER_ENTERING_WORLD", "CHAT_MSG_ADDON",
+    "GROUP_ROSTER_UPDATE" })
+
+PvPAssistant.ADDON_UPDATE_NOTIFIER.ReceivedUpdateNotification = false
+
+PvPAssistant.ADDON_UPDATE_NOTIFIER.MSG_PREFIX = "PVPA_VERCHECK"
+PvPAssistant.ADDON_UPDATE_NOTIFIER.ADDON_VERSION = C_AddOns.GetAddOnMetadata(addonName, "Version")
+-- This should maybe be moved to the CONST table as it is used a second time here: https://github.com/derfloh205/PvPAssistant/blob/b087cb9840c13d08a0199cd19ef9d2870892bf2c/Modules/MainFrame/Frames.lua#L41
+
+function PvPAssistant.ADDON_UPDATE_NOTIFIER:SendMessage()
+    local channel
+    if IsInRaid() then
+        channel = (not IsInRaid(LE_PARTY_CATEGORY_HOME) and IsInRaid(LE_PARTY_CATEGORY_INSTANCE)) and "INSTANCE_CHAT" or
+            "RAID"
+    elseif IsInGroup() then
+        channel = (not IsInGroup(LE_PARTY_CATEGORY_HOME) and IsInGroup(LE_PARTY_CATEGORY_INSTANCE)) and "INSTANCE_CHAT" or
+            "PARTY"
+    elseif IsInGuild() then
+        channel = "GUILD"
+    end
+    if channel then
+        C_ChatInfo.SendAddonMessage(self.MSG_PREFIX, self.ADDON_VERSION, channel)
+    end
+end
+
+function PvPAssistant.ADDON_UPDATE_NOTIFIER:CheckVersion(otherVersion)
+    local ownVersion = self.ADDON_VERSION:gsub("[^%d]", "")
+    otherVersion = otherVersion:gsub("[^%d]", "")
+    local updateAvailable = tonumber(otherVersion) > tonumber(ownVersion)
```

Comment:
Does this "extract" the numbers out of the addon version like
```
0.1.13 - 1.5.13 -> 0113 - 1513
0113 > 1513 
```

I think there could be some problems with this method due to cases like
`1.1.1 and 1.0.13`
wouldnt this compare
`111 > 1013` which leads to `1013` being recognized as higher than `111` ?

Try using `GUTIL:CompareVersionStrings`

Its implemented to compare each sub version

```
--- compares versions like "7.8.10" and "10.8.9" (would say right is greater then left)
---@param versionA string
---@param versionB string
---@return number result 0 if same 1 if left is greater, -1 if left is smaller
function GUTIL:CompareVersionStrings(versionA, versionB)
    local function compareSubversions(subversionA, subversionB)
        for i = 1, math.max(#subversionA, #subversionB) do
            local numA = tonumber(subversionA[i]) or 0
            local numB = tonumber(subversionB[i]) or 0
            if numA < numB then
                return -1
            elseif numA > numB then
                return 1
            end
        end
        return 0
    end

    local subversionA = strsplit(".", versionA)
    local subversionB = strsplit(".", versionB)

    local result = compareSubversions(subversionA, subversionB)

    return result
end
```

Nvm if I just misunderstood your method of comparison here!

Reply:
I didn't know there was a function for this, would probably make more Sense to use the existing function instead of using my